wagtail/core/actions/move_page.py

```diff
@@ -0,0 +1,109 @@
+import logging
+
+from django.core.exceptions import PermissionDenied
+from django.db import transaction
+from treebeard.mp_tree import MP_MoveHandler
+
+from wagtail.core.log_actions import log
+from wagtail.core.signals import post_page_move, pre_page_move
+
+
+logger = logging.getLogger("wagtail.core")
+
+
+class MovePagePermissionError(PermissionDenied):
+    """
+    Raised when the page move cannot be performed due to insufficient permissions.
+    """
+
+    pass
+
+
+class MovePageAction:
+    def __init__(self, page, target, pos=None, user=None):
+        self.page = page
+        self.target = target
+        self.pos = pos
+        self.user = user
+
+    def check(self, skip_permission_checks=False):
```

Comment:
Is it worth checking that the page can be moved to the new position based on the values of the ``sub_page_types`` and ``parent_page_types`` attributes on the parent and moved page model?